Tutorials/WinForms/Classic/create-persistent-classes-and-connect-xpo-to-database.md

```diff
@@ -0,0 +1,105 @@
+* Add the read-only `ContactName` property. XPO does not map read-only properties to database columns. Use the `FirstName` and `LastName` properties to calculate the `ContactName` property value.
+    ```csharp
+    public string ContactName {
+        get { return string.Concat(FirstName, " ", LastName); }
+    }
+    ```
+* Add the [PersistentAlias](https://docs.devexpress.com/XPO/DevExpress.Xpo.PersistentAliasAttribute) attribute to the `ContactName` property. 
+  >The [PersistentAlias](https://docs.devexpress.com/XPO/DevExpress.Xpo.PersistentAliasAttribute) attribute specifies an expression that substitutes for the property name in a SQL query. The expression passed to the attribute constructor should follow the [Criteria Language Syntax](https://docs.devexpress.com/CoreLibraries/4928/devexpress-data-library/criteria-language-syntax). 
+    ```csharp
+    [PersistentAlias("concat(FirstName, ' ', LastName)")]
+    public string ContactName {
+        get { return string.Concat(FirstName, " ", LastName); }
+    }
+    ```
+* Use the same approach to add the `Order` class with the `ProductName`(String), `OrderDate`(DateTime), and `Freight`(decimal) properties. The complete code is available here:  
+  [Order.cs](/Tutorials/WinForms/Classic/CS/DataAccess/Order.cs)  
+  [Order.vb](/Tutorials/WinForms/Classic/VB/DataAccess/Order.vb)
+* Add the `Orders` and `Customer` properties to create a [relationship](https://docs.devexpress.com/XPO/2041/concepts/relationships-between-objects) between the **Customer** and **Order** classes.
+    ```csharp
+    // Customer.cs
+    [Association("CustomerOrders")]
+    public XPCollection<Order> Orders {
+        get { return GetCollection<Order>(nameof(Orders)); }
+    }
+    
+    // Order.cs
+    private Customer fCustomer;
+    [Association("CustomerOrders")]
+    public Customer Customer {
+        get { return fCustomer; }
+        set { SetPropertyValue(nameof(Customer), ref fCustomer, value); }
+    }
+    ```
+* Add a reference to the **System.Configuration** assembly. This assembly contains classes used to access connection string settings from the application's configuration file [Connection Strings and Configuration Files](https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/connection-strings-and-configuration-files).
+* Add the *ConnectionHelper.cs* file, copy code from the example ([ConnectionHelper.cs](/Tutorials/WinForms/Classic/CS/DataAccess/ConnectionHelper.cs)), and paste it to a new file. 
+  >This code can be used without modifications in a real application. Refer to the following article for additional details: [Connecting to a Data Store](https://docs.devexpress.com/XPO/2020/feature-center/connecting-to-a-data-store).
+* Open the *Program.cs* file and add the `ConnectionHelper.Connect` method call to the `Main` method.
+  >Visual Studio does not create the *Program.vb* file in a VB.NET project, use the `Form1` class constructor, instead.
```

Comment:
```suggestion
  >Visual Studio does not create the *Program.vb* file in a VB.NET project - use the `Form1` class constructor instead.
```

Reply:
I'm not use why you wrote ", use the `Form1` class constructor, instead", and why you use a comma before instead?

Reply:
I changed these sentences to ">Visual Studio does not create the *Program.vb* file in a VB.NET project. You can add this code to the `Form1` class constructor before the `InitializeComponent` method".

47b7f58fe3cedf0b3c49ac77a06b2f29a875f0e6